### ISSUE TITLE
BUG: FutureWarning when splitting a dataframe using `np.split`

### ISSUE THREAD

**Issue by amanlai** [pandas-dev/pandas#57351]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
To reproduce it:

import numpy as np
import pandas as pd

df = pd.DataFrame({'a': [1, 2, 3]})
lst = np.split(df, 3)
```

### Issue Description

The above code raises a FutureWarning:
```none
FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
```
As far as I understand, `np.split` uses `np.swapaxes` which is raising this warning.

### Expected Behavior

Not show a warning.

### Installed Versions

<details>

python                : 3.11.5
pandas                : 2.2.0
numpy                 : 1.26.3

</details>

**Comment by VISWESWARAN1998** [comment]
`swapaxes` is called from shape_base.py in numpy package

**Comment by rhshadrach** [comment]
Thanks for the report - is there a reason you prefer to use `np.swapaxes(df)` over `df.T`?

**Comment by amanlai** [comment]
@rhshadrach the use case here is not `np.swapaxes(df)` itself, it's `np.split(df)`, which apparently uses `np.swapaxes` under the hood.

**Comment by rhshadrach** [comment]
Ah, thanks @amanlai. On main, `DataFrame.swapaxes` has been removed and the OP gives the output:

```[array([[1]]), array([[2]]), array([[3]])]```

On 2.2.x, I am seeing

```
[   a
0  1,    a
1  2,    a
2  3]
```

cc @jorisvandenbossche @phofl @mroeschke

**Comment by rhshadrach** [comment]
Marking this as a blocker for 3.0 so a decision is made.

**Comment by Aloqeely** [comment]
I don't think many people use `np.split` on a DataFrame to consider reverting this deprecation. ([Stackoverflow question](https://stackoverflow.com/questions/77576750/futurewarning-dataframe-swapaxes-is-deprecated-and-will-be-removed-in-a-futur) did not have many interactions)

And as stated in [numpy/numpy#24889 (comment)](https://github.com/numpy/numpy/issues/24889#issuecomment-1895503977) it's possible to calculate the start/stop slices and then manually slicing using `df.iloc[start:stop]`

**Comment by WillAyd** [comment]
I agree with @Aloqeely. Is there an upstream issue for NumPy on this topic? I think should be resolved there and use `df.T` like @rhshadrach suggests

**Comment by Aloqeely** [comment]
Is it sensible to implement a `DataFrame.split` function for convenience? Since `np.split` doesn't work appropriately on DataFrames anymore.

I can work on it next week if you are all ok with it.

**Comment by jorisvandenbossche** [comment]
> Is there an upstream issue for NumPy on this topic?

https://github.com/numpy/numpy/issues/24889#issuecomment-1895503977

**Comment by WillAyd** [comment]
> Is it sensible to implement a `DataFrame.split` function for convenience? Since `np.split` doesn't work appropriately on DataFrames anymore.

I don't think so - generally our goal is to reduce the footprint of our API, and I don't see this as a huge value add over the other method you have suggested:

> And as stated in [numpy/numpy#24889 (comment)](https://github.com/numpy/numpy/issues/24889#issuecomment-1895503977) it's possible to calculate the start/stop slices and then manually slicing using `df.iloc[start:stop]`

**Comment by ddelange** [comment]
> Is it sensible to implement a `DataFrame.split` function for convenience? Since `np.split` doesn't work appropriately on DataFrames anymore.
> 
> I can work on it next week if you are all ok with it.

+1 :+1: splitting a dataframe into equally sized chunks (except for the trailing chunk) is a routine task in ML and other batching applications.

requiring end-users to replace np.split/np.array_split with some bespoke `iloc` helper function sounds to me like a lot of duplicated boilerplate, wasted brain cycles over time, and overall increased bug surface for the ecosystem

**Comment by jbrockmendel** [comment]
i dont see a warning on main.  IIUC that is bc we enforced the swapaxes deprecation?  If so, i think this is closable.

**Comment by rhshadrach** [comment]
Yes, but in 2.3.x you'll get back 3 DataFrames, on main you get back 3 NumPy arrays. Still, I agree this is closable. I do wonder if `DataFrame.split` would be a useful addition to the API and better than relying on NumPy functionality.

**Comment by rhshadrach** [comment]
I opened #62719. Closing.

**Comment by ddelange** [comment]
> > Is it sensible to implement a `DataFrame.split` function for convenience? Since `np.split` doesn't work appropriately on DataFrames anymore.
> > I can work on it next week if you are all ok with it.
> 
> +1 👍 splitting a dataframe into equally sized chunks (except for the trailing chunk) is a routine task in ML and other batching applications.
> 
> requiring end-users to replace np.split/np.array_split with some bespoke `iloc` helper function sounds to me like a lot of duplicated boilerplate, wasted brain cycles over time, and overall increased bug surface for the ecosystem

with swapaxes definitively deprecated in pandas v3, my array_split helper is now the following:

```diff
- from numpy import array_split
+ from numpy import arange, array_split

- dfs = array_split(df, n_chunks, axis=axis)
+ indices = array_split(arange(df.shape[axis]), n_chunks)
+ dfs = [df.take(idx, axis=axis) for idx in indices]
```
